plugins/outputs/bigquery/bigquery.go

```diff
@@ -0,0 +1,248 @@
+
+	return nil
+}
+
+func (b *BigQuery) groupByMetricName(metrics []telegraf.Metric) map[string][]bigquery.ValueSaver {
+	groupedMetrics := make(map[string][]bigquery.ValueSaver)
+
+	for _, m := range metrics {
+		bqm := newValuesSaver(m)
+		groupedMetrics[m.Name()] = append(groupedMetrics[m.Name()], bqm)
+	}
+
+	return groupedMetrics
+}
+
+func newValuesSaver(m telegraf.Metric) *bigquery.ValuesSaver {
+	s := make(bigquery.Schema, 0)
+	r := make([]bigquery.Value, 0)
+	timeSchema := timeStampFieldSchema()
+	s = append(s, timeSchema)
+	r = append(r, m.Time())
+
+	s, r = tagsSchemaAndValues(m, s, r)
+	s, r = valuesSchemaAndValues(m, s, r)
+
+	return &bigquery.ValuesSaver{
+		Schema: s.Relax(),
+		Row:    r,
+	}
+}
+
+func timeStampFieldSchema() *bigquery.FieldSchema {
+	return &bigquery.FieldSchema{
+		Name: timeStampFieldName,
+		Type: bigquery.TimestampFieldType,
+	}
+}
+
+func tagsSchemaAndValues(m telegraf.Metric, s bigquery.Schema, r []bigquery.Value) ([]*bigquery.FieldSchema, []bigquery.Value) {
+	for _, t := range m.TagList() {
+		s = append(s, tagFieldSchema(t))
+		r = append(r, t.Value)
+	}
+
+	return s, r
+}
+
+func tagFieldSchema(t *telegraf.Tag) *bigquery.FieldSchema {
+	return &bigquery.FieldSchema{
+		Name: t.Key,
+		Type: bigquery.StringFieldType,
+	}
+}
+
+func valuesSchemaAndValues(m telegraf.Metric, s bigquery.Schema, r []bigquery.Value) ([]*bigquery.FieldSchema, []bigquery.Value) {
+	for _, f := range m.FieldList() {
+		s = append(s, valuesSchema(f))
+		r = append(r, f.Value)
+	}
+
+	return s, r
+}
+
+func valuesSchema(f *telegraf.Field) *bigquery.FieldSchema {
+	return &bigquery.FieldSchema{
+		Name: f.Key,
+		Type: valueToBqType(f.Value),
+	}
+}
+
+func valueToBqType(v interface{}) bigquery.FieldType {
+	switch reflect.ValueOf(v).Kind() {
+	case reflect.Int, reflect.Int16, reflect.Int32, reflect.Int64:
+		return bigquery.IntegerFieldType
+	case reflect.Float32, reflect.Float64:
+		return bigquery.FloatFieldType
+	case reflect.Bool:
+		return bigquery.BooleanFieldType
+	default:
+		return bigquery.StringFieldType
+	}
+}
+
+func (b *BigQuery) insertToTable(metricName string, metrics []bigquery.ValueSaver) {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, b.Timeout.Duration)
+	defer cancel()
+
+	tableName := b.metricToTable(metricName)
+	table := b.client.DatasetInProject(b.Project, b.Dataset).Table(tableName)
+	inserter := table.Inserter()
+
+	if err := inserter.Put(ctx, metrics); err != nil {
+		b.Log.Errorf("inserting metric %q failed: %v", metricName, err)
+	}
+}
+
+func (b *BigQuery) metricToTable(metricName string) string {
+	if strings.Contains(metricName, "-") {
+
+		dhm := strings.ReplaceAll(metricName, "-", b.ReplaceHyphenTo)
+
+		if warned, _ := b.warnedOnHyphens[metricName]; !warned {
+			b.Log.Warn("Metric %q contains hyphens please consider using the rename processor plugin, falling back to %q", dhm)
+			b.warnedOnHyphens[metricName] = true
+		}
+
+		return dhm
+	}
+
+	return metricName
```

Comment:
Please invert the logic to save indention levels:
```suggestion
	if ! strings.Contains(metricName, "-") {
		return metricName
	}

	dhm := strings.ReplaceAll(metricName, "-", b.ReplaceHyphenTo)
	if warned, _ := b.warnedOnHyphens[metricName]; !warned {
		b.Log.Warn("Metric %q contains hyphens please consider using the rename processor plugin, falling back to %q", dhm)
		b.warnedOnHyphens[metricName] = true
	}

	return dhm
```